cache/redis_cache_test.go

```diff
@@ -65,15 +64,50 @@ func generateRedisClientAndServer(t *testing.T) *redisCache {
 
 func TestRedisCacheAddGet(t *testing.T) {
 	c := generateRedisClientAndServer(t)
-	c1 := generateRedisClientAndServer(t)
 	defer func() {
-		c1.Close()
 		c.Close()
 	}()
-	cacheAddGetHelper(t, c, c1)
+	cacheAddGetHelper(t, c)
 }
 
 func TestRedisCacheMiss(t *testing.T) {
 	c := generateRedisClientAndServer(t)
 	cacheMissHelper(t, c)
 }
+func TestStringFromToByte(t *testing.T) {
+	c := generateRedisClientAndServer(t)
```

Comment:
It's not needed to test encode and decode functions (once they're decoupled from redis cache)

Reply:
good point. I asked myself this question before doing this test but thought having a dedicated test that only that only focus on this part would be easier for debugging if this part contains a pb  (at least for coding this part very helpful).
Even if it's redundant with other tests it's still worth keep it IMHO